podium/datasets/iterator.py

```diff
@@ -310,13 +310,12 @@ def __iter__(self) -> PythonIterator[Batch]:
             self._shuffler_state = self.get_internal_random_state()
             self._shuffler.shuffle(indices)
 
-        data = self._dataset[indices]
-
         # If iteration was stopped, continue where we left off
         start = self.iterations * self.batch_size
 
-        for i in range(start, len(data), self.batch_size):
-            batch_instances = data[i : i + self.batch_size]
+        for i in range(start, len(self._dataset), self.batch_size):
+            batch_indices = indices[i : i + self.batch_size]
+            batch_instances = self._dataset[batch_indices]
```

Comment:
How does this work with HF datasets? It does not cause the copy of the whole dataset as before? Does it only cause copying of the range? In any case, at some point, we should wrap this in a view. Leave it as-is for now, as it can be changed later without changing API.

Reply:
It doesn't, the `HFDatasetConverter` implements its own `__get__`, which essentially delegates it to the underlying arrow dataset implementation, everything stays on disk and not in memory. I'm a bit wary of how indexing vs slicing behaves speed-wise, will check this at some point.
Definitely should explore if there's a cleaner way of doing this.